northstar/src/runtime/fork/forker/impl.rs

```diff
@@ -22,18 +23,169 @@ use nix::{
+                                assert!(entry.is_none(), "PID already registered");
+
+                                exit_tx
+                                    .send(rx)
+                                    .await
+                                    .expect("Failed to send exit status future");
+                            }
+                            Ok(Some(init::Message::Exit { pid, exit_status })) => {
+                                let tx = ps.remove(&pid).expect("PID not registered");
+
+                                // TODO the process exit status is sent as the init exit status
+                                // because that's what the runtime and tests expect currently.
+                                let init_exit_status = exit_status.clone();
+
+                                tx.send(exit_status)
+                                    .expect("Failed to send process exit status");
+
+                                // check if last process exited
+                                if ps.is_empty() {
+                                    // TODO blocking
+                                    waitpid(unistd::Pid::from_raw(init_pid as i32), None).expect("Failed to wait for init");
+
+                                    init_exit_tx.send(init_exit_status).expect("Failed to send init exit status");
+                                    break;
+                                }
+                            }
+                            Ok(Some(_)) => panic!("Unexpected init message"),
+                            Ok(None) | Err(_) => {
+                                debug!("Init connection closed, exiting");
+                                break;
+                            }
+                        }
+                    }
+                    exec = exec_rx.recv() => {
+                        let Command { path, args, env, output } = exec.expect("Failed to receive command");
+                        match output {
+                            ProcessOutput::Pty { slave }=> {
+                                stream
+                                    .send(init::Message::Exec { path, args, env, pty: true })
+                                    .await
+                                    .expect("Failed to send");
+
+                                let pty =  std::fs::OpenOptions::new()
+                                    .read(true)
+                                    .write(true)
+                                    .open(slave)
+                                    .expect("Failed to open file");
+
+                                stream.send_fd(pty).await.expect("Failed to send pty fd");
+                            }
+                            ProcessOutput::Files { stdout, stderr } => {
+                                stream
+                                    .send(init::Message::Exec { path, args, env, pty: false })
+                                    .await
```

Comment:
I see that `Message::Exec` is sent in two flavours. One with a pty and one with stdio. The check in init which type it is is based on the the `pty` flag. 
Check the `Message` definition of the communication between the forker and the runtime. Replace the `pty` flag with a enum that contains `Option<File etc>`. Add a `send` method (like `forker::impl::send` that knows when to send the pty etc..